pkg/trace/writer/services.go

```diff
@@ -0,0 +1,134 @@
+package writer
+
+import (
+	"encoding/json"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics/timing"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// pathServices is the target host API path for delivering services.
+const pathServices = "/api/v0.2/services"
+
+// ServiceWriter aggregates incoming services and flushes them to the Datadog API.
+type ServiceWriter struct {
+	in      <-chan pb.ServicesMetadata
+	stats   *info.ServiceWriterInfo
+	buffer  pb.ServicesMetadata
+	senders []*sender
+	stop    chan struct{}
+}
+
+// NewServiceWriter returns a new, ready to use ServiceWriter. Run must be called before
+// sending metadata down the channel.
+func NewServiceWriter(cfg *config.AgentConfig, in chan pb.ServicesMetadata) *ServiceWriter {
+	sw := &ServiceWriter{
+		in:     in,
+		stats:  &info.ServiceWriterInfo{},
+		buffer: make(pb.ServicesMetadata),
+		stop:   make(chan struct{}),
+	}
+	sw.senders = newSenders(cfg, sw, pathServices, 10, 10)
```

Comment:
what's up with these hardcoded values instead of getting them from the config?

Reply:
😊I have no idea what to make these. We'll be removing the service writer very soon (likely next week) so I didn't place much importance on it. There's no reason to add yaml config for these. IMO we should just have fixed values or derived from other config settings.

Reply:
I set the connection limit to `2` and the queue size to `10`. Hard coded. Since this is being deprecated and is super low volume, I don't think there's any point in changing it. `2` is 2x what was there before anyway. Is that bad?

Reply:
As you say, this is so low traffic it probably hardly matters. But if you're not going to use the yaml config stuff from the services, then you shouldn't even attempt to load it up there on the config code. And if you already added the code to load and test it, you might as well plug it here :joy:

Reply:
You're right. I'm removing it. The change of this service writer being completely removed by the next release (which will include this change) is 90%.